src/spartan/sumcheck.rs

```diff
@@ -360,4 +360,94 @@ impl<G: Group> SumcheckProof<G> {
       vec![poly_A[0], poly_B[0], poly_C[0], poly_D[0]],
     ))
   }
+
+  pub fn prove_cubic_with_additive_term_batch<F>(
+    claim: &G::Scalar,
+    num_rounds: usize,
+    poly_A: &mut MultilinearPolynomial<G::Scalar>,
+    poly_B_vec: &mut Vec<MultilinearPolynomial<G::Scalar>>,
+    poly_C_vec: &mut Vec<MultilinearPolynomial<G::Scalar>>,
+    poly_D_vec: &mut Vec<MultilinearPolynomial<G::Scalar>>,
+    coeffs: &[G::Scalar],
+    comb_func: F,
+    transcript: &mut G::TE,
+  ) -> Result<(Self, Vec<G::Scalar>, G::Scalar, Vec<Vec<G::Scalar>>), NovaError>
+  where
+    F: Fn(&G::Scalar, &G::Scalar, &G::Scalar, &G::Scalar) -> G::Scalar + Sync,
+  {
+    let mut r: Vec<G::Scalar> = Vec::new();
+    let mut polys: Vec<CompressedUniPoly<G::Scalar>> = Vec::new();
+    let mut claim_per_round = *claim;
+
```

Comment:
I think there should be a check that the vectors and coefficients are all of the same length. If that's not the case, this is going to have unexpected results — which I don't think we want to take responsibility for producing.